prov/efa/src/efa_mr.c

```diff
@@ -810,14 +811,19 @@ static int efa_mr_reg_impl(struct efa_mr *efa_mr, uint64_t flags, void *attr)
 	efa_mr->mr_fid.mem_desc = NULL;
 	efa_mr->mr_fid.key = FI_KEY_NOTAVAIL;
 
-	ret = efa_mr_hmem_setup(efa_mr, mr_attr, flags);
+	memcpy(&mr_attr, attr, sizeof(struct fi_mr_attr));
```

Comment:
I think you should use `sizeof(*attr)`, `sizeof(struct fi_mr_attr)` is using the new header, which could be larger than the actual size of `attr` and cause segfault.

Reply:
I don't think `sizeof(*attr)` works because `attr` is `void *`.

I also considered this problem. memcpy does not define the behavior in this case. To be extra safe we can add FI_VERSION check and modify the size.

Reply:
but how did you get the size for an older libfabric version in the new libfabric code?

Reply:
Please see update. We can exclude hmem_data from the size.

Reply:
This is messy, I am not sure whether it is desirable. @wzamazon WDYT?

Reply:
I do not quite understand the problem it try to solve. So it appears that user requested 1.19 API but did not pass hmem_data?

Reply:
It's application use old libfabric headers to compile, but run with latest libfabric main branch

Reply:
application doesn't pass hmem_data